src/display/graphics.c

```diff
@@ -0,0 +1,387 @@
+/*
+ * \file graphics.c
+ *
+ * \brief Lower level graphical functions for screen displays. This lower level
+ * is necessary to house the system task that should not be visible to users.
+ *
+ * This file represents a low-level interface for interacting with the built-in
+ * LCD touch screen.
+ *
+ * \copyright (c) 2017, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//this is a test
+#include "display/graphics.h"
+
+#include <stdint.h>
+#include <stdlib.h>  // free
+#include <sys/_stdint.h>
+
+#include "pros/apix.h"
+#include "v5_api.h"  // vexDisplay*
+
+static mutex_t _display_mutex = NULL;
+
+void display_set_color_fg(uint32_t color) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
```

Comment:
So this may be a bad idea... but should the function body between the mutex take and grab be wrapped in a macro or similar interface?

Could be like:
```
#define MACRO_WRAPPER(x) \
  do { \
    mutex_take(_display_mutex, TIMEOUT_MAX); \
    x; \
    mutex_give(_display_mutex); \
  } while(0)

void display_do_thing(void) {
  MUTEX_WRAPPER(vexDisplayForegroundColor(color));
}
```

Reply:
Really just trying to find a way to avoid having to modify all of these calls if we ever modify them again. But also that's what find and replace works for, so keeping it as-is works totally fine for me too.

Reply:
I don't think it's worth it, we don't have anything else like this in any other of our files for mutex claiming. At the end of the day if we gotta modify all the mutex claiming / releasing calls we can just ctrl-f replace the occurences within the file.

Reply:
That works for me :+1: